proposals/async-streams.md

```diff
@@ -103,16 +103,20 @@ namespace System.Collections.Generic
 ```
 TryGetNext is used in an inner loop to consume items with a single interface call as long as they're available synchronously.  When the next item can't be retrieved synchronously, it returns false, and any time it returns false, a caller must subsequently invoke WaitForNextAsync to either wait for the next item to be available or to determine that there will never be another item. Typical consumption (without additional language features) would look like:
 ```C#
-IAsyncEnumerable<T> enumerator = enumerable.GetAsyncEnumerator();
-while (await enumerator.WaitForNextAsync())
+IAsyncEnumerator<T> enumerator = enumerable.GetAsyncEnumerator();
+try
 {
-    while (true)
+    while (await enumerator.WaitForNextAsync())
```

Comment:
Still think this shouldn't use the work `Wait` as that's normally blocking. 

`await enumerator.WhenNextAsync()`?

E.g. `Task.Wait` vs `Task.When`